### PR TITLE
[Workspace] Validate if workspace exists when setup inside a workspace

### LINKED ISSUE #6155: [Workspace] Validate the workspace id in the url
**Is your feature request related to a problem? Please describe.**

For now, workspace plugin will take whatever in url path if the path matches the pattern of workspace, there should be a validation to check if the workspace in url is valid.

**Describe the solution you'd like**

Check when workspace start to setup and gives an friendly error page when user can not get access to the specific workspace.

**Describe alternatives you've considered**

**Additional context**

#4944

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

When user enters a url with workspaceId pattern inside url, workspace plugin should validate if the id exists, or should show an friendly error page to ask user to go back to home page without a pattern in url.

### Issues Resolved

closes #6155 

## Screenshot

image

## Testing the changes

- Using the branch to bootstrap
- Enable workspace feature flag in yml file by set `workspace.enabled: true`
- Start OSD by using `yarn start --no-base-path` to make sure no random base path will be appended
- Navigate to devTools
- Insert test workspaces by calling
```
PUT .kibana/_doc/workspace:foo
{
  "type": "workspace",
  "workspace": {
    "name": "foo"
  }
}
```
- Visit devTools under the test workspace: `http://localhost:5601/w/foo/app/dev_tools`, the page should render successfully.
- Visit devTools under a unexisting workspace: `http://localhost:5601/w/not_existing/app/dev_tools`, should see a page with error just like the screenshot above.
- Visit the error page under a existing worksapce: `http://localhost:5601/w/foo/app/workspace_fatal_error`, user should be redirect to `http://localhost:5601/w/foo/app/workspace_overview` page, though the page has not been implement yet.

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/application.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { OpenSearchDashboardsContextProvider } from '../../opensearch_dashboards_react/public';
+import { WorkspaceFatalError } from './components/workspace_fatal_error';
+import { Services } from './types';
+
+export const renderFatalErrorApp = (params: AppMountParameters, services: Services) => {
+  const { element } = params;
+  const history = params.history as ScopedHistory<{ error?: string }>;
+  ReactDOM.render(
+    <OpenSearchDashboardsContextProvider services={services}>
+      <WorkspaceFatalError error={history.location.state.error} />
```
Comment (Flyingliuhub) [inline src/plugins/workspace/public/application.tsx:18]: if error is null, should we consider `const error = history.location.state?.error;` to ensure error exists before accessing it, then it won't throw an error when rendering the component.

File: src/plugins/workspace/public/plugin.ts [hunk 2]
```diff
@@ -33,9 +43,60 @@ export class WorkspacePlugin implements Plugin<{}, {}, {}> {
...
+         * Fatal error service does not support customized actions
+         * So we have to use a self-hosted page to show the errors and redirect.
+         */
+        (async () => {
+          const [{ application, chrome }] = await core.getStartServices();
+          chrome.setIsVisible(false);
+          application.navigateToApp(WORKSPACE_FATAL_ERROR_APP_ID, {
+            replace: true,
+            state: {
+              error: result.error,
```
Comment (Flyingliuhub) [inline src/plugins/workspace/public/plugin.ts:58]: do we consider to to check result.error exists before access it. like as `const error = result.error || "Unknown error";` so ensures that `error` always have value

**Review by Flyingliuhub — APPROVED** [review]

File: src/plugins/workspace/public/plugin.ts (hunk 2 above)
Comment (SuZhou-Joe) [reply src/plugins/workspace/public/plugin.ts]: If the error is empty or null, we will hide the callout component to not display the detailed error message, but update to `result?.error` to protect the code.

File: src/plugins/workspace/public/application.tsx (hunk 1 above)
Comment (SuZhou-Joe) [reply src/plugins/workspace/public/application.tsx]: Sure, done for that.

File: src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import { FormattedMessage } from '@osd/i18n/react';
+import { IBasePath } from 'opensearch-dashboards/public';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { formatUrlWithWorkspaceId } from '../../../../../core/public/utils';
+
+export function WorkspaceFatalError(props: { error?: string }) {
+  const {
+    services: { application, http },
+  } = useOpenSearchDashboards();
+  const goBackToHome = () => {
```
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx:24]: Should it simply go back to `/` which might  be cleaner?

File: src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                  id="core.fatalErrors.somethingWentWrongTitle"
+                  defaultMessage="Something went wrong"
+                />
+              </h2>
+            }
+            body={
+              <p>
+                <FormattedMessage
+                  id="core.fatalErrors.tryGoBackToDefaultWorkspaceDescription"
+                  defaultMessage="The workspace you want to go can not be found, try go back to home."
```
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx:50]: Nit:
```suggestion
                  defaultMessage="The workspace you are trying to access cannot be found. Please return to the homepage and try again."
```

File: src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                />
+              </p>
+            }
+            actions={[
+              <EuiButton color="primary" fill onClick={goBackToHome}>
+                <FormattedMessage
+                  id="core.fatalErrors.goBackToHome"
+                  defaultMessage="Go back to home"
```
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx:58]:
```suggestion
                  defaultMessage="Go back to homepage"
```

File: src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx (hunk 3 above)
Comment (SuZhou-Joe) [reply src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx:24]: In order to go back to homepage of default workspace, we need to:
1. Remove the workspace id inside the url path. (which will be done by second parameter of `formatUrlWithWorkspaceId`)
2. construct the path with potential basePath defined in yml file.

So I think the call to formatUrlWithWorkspaceId is necessary here.

Comment (ruanyl) [reply src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx:24]: If we let it navigate to `/<basePath>`, then it should be able to automatically redirect to landing page, which by default is the `home` app, what do you think?

Comment (SuZhou-Joe) [reply src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx:23]: Make sense, we should take `default route` into consideration.Thanks for catching this and updated accordingly.

**Review by ruanyl — APPROVED** [review]

**Pull request merged**